tgui/packages/tgui/interfaces/Hypertorus.js

```diff
@@ -0,0 +1,302 @@
+  const fusionMax = Math.max(1, ...fusion_gases.map(gas => gas.amount));
+  const moderatorMax = Math.max(1, ...moderator_gases.map(gas => gas.amount));
+  return (
+    <Window
+      width={500}
+      height={600}
+      scrollable
+      resizable
+      title="Fusion Reactor">
+      <Window.Content>
+        <Section title="Switches">
+          <Flex m={-0.5}>
+            <Flex.Item m={0.5} color="label">
+              {'Start power: '}
+              <Button
+                disabled={data.power_level > 0}
+                icon={data.start_power ? 'power-off' : 'times'}
+                content={data.start_power ? 'On' : 'Off'}
+                selected={data.start_power}
+                onClick={() => act('start_power')} />
+            </Flex.Item>
+            <Flex.Item m={0.5} color="label">
+              {'Start cooling: '}
+              <Button
+                disabled={start_fuel === 1
+                    || start_power === 0
+                    || data.power_level > 0}
+                icon={data.start_cooling ? 'power-off' : 'times'}
+                content={data.start_cooling ? 'On' : 'Off'}
+                selected={data.start_cooling}
+                onClick={() => act('start_cooling')} />
+            </Flex.Item>
+            <Flex.Item m={0.5} color="label">
+              {'Start fuel injection: '}
+              <Button
+                disabled={start_power === 0
+                    || start_cooling === 0}
+                icon={data.start_fuel ? 'power-off' : 'times'}
+                content={data.start_fuel ? 'On' : 'Off'}
+                selected={data.start_fuel}
+                onClick={() => act('start_fuel')} />
+            </Flex.Item>
+          </Flex>
+        </Section>
+        <Section>
+          <Section title="Internal Fusion Gases">
+            {fusion_gases.map(gas => (
+              <LabeledList.Item
+                key={gas.name}
+                label={getGasLabel(gas.name)}>
+                <ProgressBar
+                  color={getGasColor(gas.name)}
+                  value={gas.amount}
+                  minValue={0}
+                  maxValue={fusionMax}>
+                  {toFixed(gas.amount, 2) + ' moles'}
+                </ProgressBar>
+              </LabeledList.Item>
+            ))}
+          </Section>
+          <Section title="Moderator Gases">
+            {moderator_gases.map(gas => (
+              <LabeledList.Item
+                key={gas.name}
+                label={getGasLabel(gas.name)}>
+                <ProgressBar
+                  color={getGasColor(gas.name)}
+                  value={gas.amount}
+                  minValue={0}
+                  maxValue={moderatorMax}>
+                  {toFixed(gas.amount, 2) + ' moles'}
+                </ProgressBar>
+              </LabeledList.Item>
+            ))}
+          </Section>
+        </Section>
+        <Section title="Reactor Parameters">
+          <LabeledList.Item label="Power Level">
```

Comment:
**`LabeledList` structure violation**

Wrap the whole list, not each of them individually.